core/shared/src/main/scala/fs2/concurrent/Channel.scala

```diff
@@ -176,7 +176,7 @@ object Channel {
                       State(values, size, None, (a, producer) :: producers, close),
                       signalClosure.whenA(close) *>
                         notifyStream(waiting).as(rightUnit) <*
-                        waitOnBound(producer, poll)
+                        waitOnBound(producer, poll).whenA(!close)
```

Comment:
Hmm, I'm not sure this change makes sense. `send` is supposed to block if there is no capacity in the `Channel`, and this is where the blocking happens. Semantically I would expect `sendAndClose(...)` to behave like `send(...) *> close` except atomically.